src/Core/src/Handlers/Window/WindowHandler.iOS.cs

```diff
@@ -1,17 +1,38 @@
 ﻿using System;
-using ObjCRuntime;
+using Foundation;
 using UIKit;
 
 namespace Microsoft.Maui.Handlers
 {
 	public partial class WindowHandler : ElementHandler<IWindow, UIWindow>
 	{
+		readonly WindowProxy _proxy = new();
+
 		protected override void ConnectHandler(UIWindow platformView)
 		{
 			base.ConnectHandler(platformView);
 
-			UpdateVirtualViewFrame(platformView);
+			// For newer Mac Catalyst versions, we want to wait until we get effective window dimensions from the platform.
+			if (OperatingSystem.IsMacCatalyst() && OperatingSystem.IsIOSVersionAtLeast(16))
```

Comment:
Why not `IsMacCatalystVersionAtLeast(16)`? Does this apply to all mac catalyst and to iOS 16+? The comment says it is just for new mac catalyst? I know we are super consistent, but we are trying to be a bit better with this.